common/client-libs/gateway-client/src/client.rs

```diff
@@ -517,9 +553,13 @@ impl GatewayClient {
         if !self.authenticated {
             return Err(GatewayClientError::NotAuthenticated);
         }
-        #[cfg(feature = "coconut")]
-        if self.estimate_required_bandwidth(&packets) < self.bandwidth_remaining {
-            return Err(GatewayClientError::NotEnoughBandwidth);
+        if self.estimate_required_bandwidth(&packets) > self.bandwidth_remaining {
```

Comment:
Those are all very good points. I 100% support implementing them in the future, especially since they are valid for both types of credentials. I'll put up a user story to back it up: https://github.com/orgs/nymtech/projects/1#card-72234182